overlord/snapstate/refreshhints_test.go

```diff
@@ -109,14 +112,19 @@ func (s *refreshHintsTestSuite) SetUpTest(c *C) {
 
 	s.state.Set("refresh-privacy-key", "privacy-key")
 
-	s.restoreModel = snapstatetest.MockDeviceModel(DefaultModel())
+	restoreModel := snapstatetest.MockDeviceModel(DefaultModel())
+	s.AddCleanup(restoreModel)
+	restoreEnforcedValidationSets := snapstate.MockEnforcedValidationSets(func(st *state.State) (*snapasserts.ValidationSets, error) {
+		return nil, nil
+	})
+	s.AddCleanup(restoreEnforcedValidationSets)
 }
 
 func (s *refreshHintsTestSuite) TearDownTest(c *C) {
+	s.BaseTest.TearDownTest(c)
 	dirs.SetRootDir("/")
 	snapstate.CanAutoRefresh = nil
 	snapstate.AutoAliases = nil
```

Comment:
nitpick: these can also be do with AddCleanup I suppose and drrop the TearDownTest?